QRCoder/QRCodeGenerator.cs

```diff
@@ -223,19 +302,27 @@ private static QRCodeData GenerateQrCode(BitArray bitArray, ECCLevel eccLevel, i
         // fills the bit array with a repeating pattern to reach the required length
         void PadData()
         {
-            var dataLength = eccInfo.TotalDataCodewords * 8;
+            var dataLength = eccInfo.TotalDataBits;
             var lengthDiff = dataLength - bitArray.Length;
             if (lengthDiff > 0)
             {
                 // set 'write index' to end of existing bit array
                 var index = bitArray.Length;
                 // extend bit array to required length
                 bitArray.Length = dataLength;
-                // pad with 4 zeros (or less if lengthDiff < 4)
-                index += Math.Min(lengthDiff, 4);
+                // compute padding length
+                var padLength = version > 0 ? 4 :
+                    version == -1 ? 3 :
+                    version == -2 ? 5 :
+                    version == -3 ? 7 : 9;
```

Comment:
W/ pattern matching the JIT produces a tad better code similar to a jump-table instead of a series of branches.
```suggestion
                int padLength = version switch
                {
                    > 0 => 4,
                    -1 => 3,
                    -2 => 5,
                    -3 => 7,
                    _ => 9
                };
```